extra/import_data.py

```diff
@@ -16,34 +17,34 @@
 
 def import_from_alissa(alissa_client, start_time, logger):
     '''Main function, for extracting the data from the Alissa output and send to function to parse to MongoDB'''
-     # connection with MongoDB and the correct database "vus" and collection "lastUpdatedOn"
+    # connection with MongoDB and the correct database "vus" and collection "lastUpdatedOn"
     mongo_client = pymongo.MongoClient(config.mongodb_localhost)
     db = mongo_client.vus.lastUpdatedOn
-    
+
     # Retrieve the last time MongoDB has updated itself, to filter the latest Alissa analyses only
-    last_updated_on_mongoDB = db.find_one({"version" : 1}, {"lastUpdatedOn":1, "_id":0})
-    #If not exist yet, use '2018-01-01T00:00:00.000+0000' and add to mongo (first time use)
+    last_updated_on_mongoDB = db.find_one({"version": 1}, {"lastUpdatedOn": 1, "_id": 0})
+    # If not exist yet, use '2018-01-01T00:00:00.000+0000' and add to mongo (first time use)
     if last_updated_on_mongoDB is None:
-        db.replace_one({"version" : 1}, {"lastUpdatedOn" : '2018-01-01T00:00:00.000+0000', "version" : 1}, True)
-        last_updated_on_mongoDB = db.find_one({"version" : 1}, {"lastUpdatedOn":1, "_id":0})
-    
+        db.replace_one({"version": 1}, {"lastUpdatedOn": '2018-01-01T00:00:00.000+0000', "version": 1}, True)
+        last_updated_on_mongoDB = db.find_one({"version": 1}, {"lastUpdatedOn": 1, "_id": 0})
+
     last_updated_on_mongoDB = last_updated_on_mongoDB['lastUpdatedOn']
-    
+
     # retrieving data from Alissa
     analysis = None
-    for analysis in alissa_client.get_analyses(status='COMPLETED', 
-                                        lastUpdatedAfter=last_updated_on_mongoDB, # format is '2020-01-01T00:00:00.000+0000'
-                                        analysisPipelineName='ONB01',
-                                        analysisType='INHERITANCE'): #add for testing [:x], where x is numer of iterations
-        if analysis['classificationTreeName']:
+    for analysis in alissa_client.get_analyses(status='COMPLETED',
+                                               lastUpdatedAfter=last_updated_on_mongoDB,  # format is '2020-01-01T00:00:00.000+0000'
+                                               analysisPipelineName='ONB01',
+                                               analysisType='INHERITANCE'):  # add for testing [:x], where x is numer of iterations
+        if analysis['classificationTreeName'] and alissa_client.get_analysis_report(analysis['id']):
             # retrieve basic info from Alissa about the analysis
-            patient_dn_no = alissa_client.get_report(analysis['id'])[0]['reportName'].split("_")[0]
+            patient_dn_no = alissa_client.get_analysis_report(analysis['id'])[0]['reportName'].split("_")[0]
```

Comment:
I added `alissa_client.get_analysis_report(analysis['id'])` in the if statement recently. I think this might be due to the test server of Alissa. However I encountered one old analysis, where the line `patient_dn_no = alissa_client.get_analysis_report(analysis['id'])[0]['reportName'].split("_")[0]` crashed, because Alissa could not find the analysis (which is weird). then I found out that some analysis did have an empty analysisID. this should not be possible in my opinion, so by adding it in the if statement I can skip that analysis.

I do not think it will me easier to understand to try to get this in one statement. Unless you have a suggestion on how to solve this problem differently?

Reply:
After discussion on Teams I see that your issue was mainly retrieving the same information from Alissa (database) twice. I have now changed it to use a variable instead.